src/Tizen.Account.OAuth2/Tizen.Account.OAuth2/Authorizer.cs

```diff
@@ -98,83 +103,98 @@ private TokenResponse GetRefreshToken(IntPtr requestHandle)
+            IntPtr responseCompletionId = usrData;
+            TaskCompletionSource <TokenResponse> responseCompletionSource = _taskResponseMap[responseCompletionId];
+            _taskResponseMap.Remove(responseCompletionId);
 
-                    var token = new AccessToken();
-                    token.Token = (accessToken == IntPtr.Zero) ? null : Marshal.PtrToStringAnsi(accessToken);
-                    token.TokenType = (tokenType == IntPtr.Zero) ? null : Marshal.PtrToStringAnsi(tokenType);
-                    token.Scope = scopes;
-                    token.ExpiresIn = expiresIn;
+            Interop.Response.GetError(responseHandle, out error);
+            if (error != IntPtr.Zero)
+            {
+                Log.Error(ErrorFactory.LogTag, "Error occured");
+                responseCompletionSource.SetException(ErrorFactory.GetException(error));
+            }
+            else
+            {
+                IntPtr accessToken;
+                ret = Interop.Response.GetAccessToken(responseHandle, out accessToken);
+                if (ret != (int)OAuth2Error.None)
+                {
+                    Log.Error(ErrorFactory.LogTag, "Interop failed");
+                    responseCompletionSource.SetException(ErrorFactory.GetException(ret));
+                }
 
-                    response = new TokenResponse(responseHandle);
-                    response.AccessToken = token;
-                    response.RefreshToken = (refreshToken == IntPtr.Zero) ? null : new RefreshToken() { Token = Marshal.PtrToStringAnsi(refreshToken) };
+                IntPtr tokenType;
+                ret = Interop.Response.GetTokenType(responseHandle, out tokenType);
+                if (ret != (int)OAuth2Error.None)
+                {
+                    Log.Error(ErrorFactory.LogTag, "Failed to get token type");
                 }
-            };
 
-            ret = Interop.Manager.RefreshAccessToken(_managerHandle, requestHandle, accessTokenCb, IntPtr.Zero);
-            Interop.Request.Destroy(requestHandle);
-            if (ret != (int)OAuth2Error.None || error != IntPtr.Zero)
-            {
-                if (error != IntPtr.Zero)
+                long expiresIn;
+                ret = Interop.Response.GetExpiresIn(responseHandle, out expiresIn);
+                if (ret != (int)OAuth2Error.None)
                 {
-                    throw ErrorFactory.GetException(error);
+                    Log.Error(ErrorFactory.LogTag, "Failed to get expires in");
                 }
-                else
+
+                IntPtr refreshToken;
+                ret = Interop.Response.GetRefreshToken(responseHandle, out refreshToken);
+                if (ret != (int)OAuth2Error.None)
                 {
                     Log.Error(ErrorFactory.LogTag, "Interop failed");
-                    throw ErrorFactory.GetException(ret);
+                    responseCompletionSource.SetException(ErrorFactory.GetException(ret));
```

Comment:
```suggestion
                    responseCompletionSource.SetException(ErrorFactory.GetException(ret));
                    return;
```
You should return when exception was set on responseCompletionSource